provider/datagen/src/transform/segmenter/mod.rs

```diff
@@ -255,6 +261,56 @@ fn generate_rule_break_data(
                         continue;
                     }
 
+                    // InCB* isn't a part of grapheme break propery.
+                    // See https://unicode.org/reports/tr44/#Indic_Conjunct_Break
+                    if p.name == "InCBConsonant" || p.name == "InCBLinker" || p.name == "InCBExtend"
```

Comment:
thought: this derivation will probably change relatively often, and this means there's one more part of the code to tweak every Unicode release.

I'm looking at icuexportdata and I don't see InCB in it (even in the released version I downloaded). @sffc do you know why it's not included? Do we not include derived props?


Even if we do not choose to turn this into a property, we should at the very least construct a CPT from InCB.toml _here_ and use it.

Reply:
Is this a merged property similar to General Category Group?

Reply:
No, it's a DerivedCoreProperty

https://unicode.org/reports/tr44/#Indic_Conjunct_Break

Reply:
Hmm, how stable is that formula for computing Indic_Conjunct_Break?

My first reaction is that we should export a function/class that depends on the existing data (looks like it needs Script and Indic_Syllabic_Category?), and then we can use it from here in segmenter datagen.

Please note that constructing CPTs is not cheap since it uses the WASM or ICU4C code path. We may want a class that uses a runtime algorithm.

Reply:
> Hmm, how stable is that formula for computing Indic_Conjunct_Break?

It is not stable at all, we are currently actively in discussion for changing it.

> Please note that constructing CPTs is not cheap since it uses the WASM or ICU4C code path. We may want a class that uses a runtime algorithm.

Yeah but we only have to do it once here. We'd have to do it anyway if we chose to make the property public; I'm just proposing a solution that doesn't change the public properties API.

Reply:
seems like ICU4C would be happy to add the API, someone just needs to do the work.

so for the purpose of this PR this is fine but in the medium term we should fix icu4c and icuexportdata

Reply:
I want to have a discussion, including with ICU4C, on whether this and similar properties should be exported as data (maybe a bit simpler through the stack) or whether they should always be derived at runtime from existing data (sharing more data but maybe slower).

Reply:
That's fair.

I don't think that blocks anything I want to do here, though. I am not proposing we add InCB as a public ICU4X property with data (at the moment), and for ICU4C the norm _so far_ has been that all derived properties have ICU4C data and APIs, I don't think it makes sense to block adding a uchar.h API on deciding whether the API is computing or automagical, and as far as I can tell any such uchar.h API can have its implementation swapped out rather easily.

I'll also note that all core derived properties currently have ICU4X data and APIs as well, except for InCB. It definitely makes sense to try and winnow data here if possible, a fair number of these are simple general category masks (others invoke normalization and such, that's probably worth precomputing). The tricky thing is figuring out if it's worth adding alternate codepaths to CodePointSetData and breaking the not-guaranteed but useful invariant that CodePointSetData is always backed by a CodePointInversionList.

Reply:
Another point that was brought up in discussion yesterday was that the algorithmic approach can sometimes get out-of-sync with the data published by Unicode, in addition to being subject to change over time. It seems the source of truth for property values is the UCD, not the spec algorithm.

Given all this background I'm okay adding it as a traditional property with a traditional data pipeline.

Reply:
Yes. I forgot to mention this yesterday but the algorithms are not even maintained in ICU4C, they're maintained in unicodetools.

Reply:
explicit note: this isn't a blocking concern